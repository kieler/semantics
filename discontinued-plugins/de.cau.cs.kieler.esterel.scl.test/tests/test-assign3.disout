module test_assign2:
function mynot(boolean) : boolean;
output A : boolean;
output B : boolean;
output C : boolean;

  loop
    pause;
    emit C(true);
    pause;
    emit C(true);
    pause;
    emit C(false);
    pause
  end loop
||
  var c : boolean in
    loop
      c := mynot(?C);
      emit A(c);
      pause
    end loop
  end var
||
  var value : boolean in
    value := true;
    loop
      abort
        loop
          pause
        end loop
      when A;
      value := ?A;
      emit B(value)
    end loop
  end var

end module
