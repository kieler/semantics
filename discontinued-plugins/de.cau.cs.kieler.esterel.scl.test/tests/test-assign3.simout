   0 A=1 (1) B=0 C=0 
   1 A=1 (0) B=1 (0) C=1 (1) 
   2 A=1 (0) B=1 (0) C=1 (1) 
   3 A=1 (1) B=1 (1) C=1 (0) 
   4 A=1 (1) B=1 (1) C=0 
   5 A=1 (0) B=1 (0) C=1 (1) 
   6 A=1 (0) B=1 (0) C=1 (1) 
   7 A=1 (1) B=1 (1) C=1 (0) 
   8 A=1 (1) B=1 (1) C=0 
   9 A=1 (0) B=1 (0) C=1 (1) 
  10 A=1 (0) B=1 (0) C=1 (1) 
  11 A=1 (1) B=1 (1) C=1 (0) 
  12 A=1 (1) B=1 (1) C=0 
  13 A=1 (0) B=1 (0) C=1 (1) 
  14 A=1 (0) B=1 (0) C=1 (1) 
  15 A=1 (1) B=1 (1) C=1 (0) 
  16 A=1 (1) B=1 (1) C=0 
  17 A=1 (0) B=1 (0) C=1 (1) 
  18 A=1 (0) B=1 (0) C=1 (1) 
  19 A=1 (1) B=1 (1) C=1 (0) 
  20 A=1 (1) B=1 (1) C=0 
  21 A=1 (0) B=1 (0) C=1 (1) 
  22 A=1 (0) B=1 (0) C=1 (1) 
  23 A=1 (1) B=1 (1) C=1 (0) 
  24 A=1 (1) B=1 (1) C=0 
  25 A=1 (0) B=1 (0) C=1 (1) 
  26 A=1 (0) B=1 (0) C=1 (1) 
  27 A=1 (1) B=1 (1) C=1 (0) 
  28 A=1 (1) B=1 (1) C=0 
  29 A=1 (0) B=1 (0) C=1 (1) 
  30 A=1 (0) B=1 (0) C=1 (1) 
  31 A=1 (1) B=1 (1) C=1 (0) 
  32 A=1 (1) B=1 (1) C=0 
  33 A=1 (0) B=1 (0) C=1 (1) 
  34 A=1 (0) B=1 (0) C=1 (1) 
  35 A=1 (1) B=1 (1) C=1 (0) 
  36 A=1 (1) B=1 (1) C=0 
  37 A=1 (0) B=1 (0) C=1 (1) 
  38 A=1 (0) B=1 (0) C=1 (1) 
  39 A=1 (1) B=1 (1) C=1 (0) 
  40 A=1 (1) B=1 (1) C=0 
  41 A=1 (0) B=1 (0) C=1 (1) 
  42 A=1 (0) B=1 (0) C=1 (1) 
  43 A=1 (1) B=1 (1) C=1 (0) 
  44 A=1 (1) B=1 (1) C=0 
  45 A=1 (0) B=1 (0) C=1 (1) 
  46 A=1 (0) B=1 (0) C=1 (1) 
  47 A=1 (1) B=1 (1) C=1 (0) 
  48 A=1 (1) B=1 (1) C=0 
  49 A=1 (0) B=1 (0) C=1 (1) 
  50 A=1 (0) B=1 (0) C=1 (1) 
  51 A=1 (1) B=1 (1) C=1 (0) 
  52 A=1 (1) B=1 (1) C=0 
  53 A=1 (0) B=1 (0) C=1 (1) 
  54 A=1 (0) B=1 (0) C=1 (1) 
  55 A=1 (1) B=1 (1) C=1 (0) 
  56 A=1 (1) B=1 (1) C=0 
  57 A=1 (0) B=1 (0) C=1 (1) 
  58 A=1 (0) B=1 (0) C=1 (1) 
  59 A=1 (1) B=1 (1) C=1 (0) 
  60 A=1 (1) B=1 (1) C=0 
  61 A=1 (0) B=1 (0) C=1 (1) 
  62 A=1 (0) B=1 (0) C=1 (1) 
  63 A=1 (1) B=1 (1) C=1 (0) 
  64 A=1 (1) B=1 (1) C=0 
  65 A=1 (0) B=1 (0) C=1 (1) 
  66 A=1 (0) B=1 (0) C=1 (1) 
  67 A=1 (1) B=1 (1) C=1 (0) 
  68 A=1 (1) B=1 (1) C=0 
  69 A=1 (0) B=1 (0) C=1 (1) 
  70 A=1 (0) B=1 (0) C=1 (1) 
  71 A=1 (1) B=1 (1) C=1 (0) 
  72 A=1 (1) B=1 (1) C=0 
  73 A=1 (0) B=1 (0) C=1 (1) 
  74 A=1 (0) B=1 (0) C=1 (1) 
  75 A=1 (1) B=1 (1) C=1 (0) 
  76 A=1 (1) B=1 (1) C=0 
  77 A=1 (0) B=1 (0) C=1 (1) 
  78 A=1 (0) B=1 (0) C=1 (1) 
  79 A=1 (1) B=1 (1) C=1 (0) 
  80 A=1 (1) B=1 (1) C=0 
  81 A=1 (0) B=1 (0) C=1 (1) 
  82 A=1 (0) B=1 (0) C=1 (1) 
  83 A=1 (1) B=1 (1) C=1 (0) 
  84 A=1 (1) B=1 (1) C=0 
  85 A=1 (0) B=1 (0) C=1 (1) 
  86 A=1 (0) B=1 (0) C=1 (1) 
  87 A=1 (1) B=1 (1) C=1 (0) 
  88 A=1 (1) B=1 (1) C=0 
  89 A=1 (0) B=1 (0) C=1 (1) 
  90 A=1 (0) B=1 (0) C=1 (1) 
  91 A=1 (1) B=1 (1) C=1 (0) 
  92 A=1 (1) B=1 (1) C=0 
  93 A=1 (0) B=1 (0) C=1 (1) 
  94 A=1 (0) B=1 (0) C=1 (1) 
  95 A=1 (1) B=1 (1) C=1 (0) 
  96 A=1 (1) B=1 (1) C=0 
  97 A=1 (0) B=1 (0) C=1 (1) 
  98 A=1 (0) B=1 (0) C=1 (1) 
  99 A=1 (1) B=1 (1) C=1 (0) 
