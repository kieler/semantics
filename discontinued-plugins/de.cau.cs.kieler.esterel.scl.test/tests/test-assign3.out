module test_assign2:
function mynot(boolean) : boolean;
output A : boolean;
output B : boolean;
output C : boolean;

  loop
    pause;
    emit C(true);
    pause;
    emit C(true);
    pause;
    emit C(false);
    pause
  end loop
||
  var c : boolean in
    loop
      c := mynot(?C);
      emit A(c);
      pause
    end loop
  end var
||
  var value := true : boolean in
    loop
      await A;
      value := ?A;
      emit B(value)
    end loop
  end var

end module
